deepspeed/__init__.py

```diff
@@ -337,6 +337,14 @@ def init_inference(model, config=None, **kwargs):
             raise ValueError(f"Conflicting argument '{key}' in 'config':{config_dict[key]} and kwargs:{kwargs[key]}")
     config_dict.update(kwargs)
 
+    # Set the number of weight quantization groups if an optional 'quantize_groups' argument is given
+    if "quantize_groups" in config_dict:
+        if not ("dtype", torch.int8) in config_dict.items():
+            raise ValueError("'dtype' argument expected int8 when 'quantize_groups' argument is provided")
+        quant = QuantizationConfig()
+        quant.weight.q_groups = config_dict.pop("quantize_groups")
+        config_dict["quant"] = quant
+
```

Comment:
I believe you are adding `quantize_groups` as a shortcut for the admittedly convoluted current quantize config settings? For example, with this you could just pass `quantize_groups=2` rather than `quant={"weight":{"q_groups":2}}`.

But perhaps we should look into how we can simplify the quantize settings or at the very least add this logic to the `DeepSpeedInferenceConfig` class as a pydantic validator (so that the config logic is consolidated there).